### PR TITLE
adds aten.eq.scalar/tensor -> sdfg

### PR DESCRIPTION
**Pull request by WashimNeupane**
## Description
Adds support for the aten.eq.scalar and aten.eq.tensor operation to the pytorch frontend.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor / internal change (no functional change)
- [ ] Documentation update

## Quality Checklist
*Please ensure the following are completed before requesting review:*

* [x] Runtime-compatible (no externally visible existing method has its parameters changed, was deleted or has its exception behavior changed)
* [x] Compile-time compatible (new arguments & properties added have default values, but users need to recompile against the new version to work)

IR:
* [ ] Introduces new types into the SDFG IR
* [ ]  Changed / added properties on SDFG IR Elements (LibraryNodes, ControlFlowNodes DataFlowNodes etc.)
  *  [ ] New / modified types have serialization / deserialization support
  *  [ ] Deserialization is backward compatible (new serialization can deserialize old format and represent it in the new format if applicable)
  *  [ ] Serialized format is backward compatible (previous deserializer can read new serialized format without incorrectness (just lacking additional guarantees that are optional or did not exist before)
  *  [ ] Serializer has an option to emit the previous format

### Unit Tests
- [x] New unit tests have been added for the changes.
- [x] Existing unit tests pass locally.
- [x] Edge cases and error paths are covered.

### Integration Tests
- [ ] New integration tests have been added (or existing ones updated).
- [ ] Integration tests pass locally.
- [ ] Interactions with other components/services are verified.

### Documentation
- [ ] Public APIs, options, and behavior changes are documented.
- [ ] README / docs site / inline comments updated where applicable.
- [ ] Changelog / release notes updated (if applicable).

### Test Coverage
- [ ] Test coverage has not decreased as a result of this change.
- [ ] Coverage report reviewed and acceptable.
- [ ] Critical / new code paths are covered.

## How Has This Been Tested?
Describe the tests you ran to verify your changes and provide instructions to reproduce.
Include relevant details for your test configuration (OS, Python/LLVM/Docc version, hardware/target).

## Additional Context
Add any other context about the PR here (design decisions, trade-offs, follow-ups).

### CODE REVIEW

File: python/bindings/builder/py_structured_sdfg_builder.cpp [hunk 1]
```diff
@@ -1026,35 +1028,34 @@ void PyStructuredSDFGBuilder::add_elementwise_op(
...
-            prim_type = A_type.primitive_type();
-        } else if (is_A_signed_int && is_B_unsigned_int) {
-            code_type = UnsignedInteger;
-            prim_type = B_type.primitive_type();
-        } else if (is_A_signed_int && is_B_signed_int) {
-            code_type = SignedInteger;
-            prim_type = A_type.primitive_type();
-        } else {
-            is_scalar_op = false;
-        }
```
Comment (Moehre2) [inline python/bindings/builder/py_structured_sdfg_builder.cpp:1058]: I do not get, why you made that change?

File: python/bindings/builder/py_structured_sdfg_builder.cpp
Comment (Moehre2) [inline python/bindings/builder/py_structured_sdfg_builder.cpp:1]: The `PyStructuredSDFGBuilder::add_elementwise_op` is for creating specialized nodes, e.g., `AddNode`, `SubNode` etc... Create a new method called `add_elementwise_tasklet_op` and give it a variable number of inputs and input tensors, as well as a `TaskletCode`. Also register it in [python/bindings/bindings.cpp](https://github.com/daisytuner/docc/tree/add-comparision-ops/python/bindings/bindings.cpp).

This way we have a bindings method that can create tasklet tensor nodes with any tasklet code. See [PyStructuredSDFGBuilder::add_elementwise_cmath_op](https://github.com/daisytuner/docc/blob/add-comparision-ops/python/bindings/builder/py_structured_sdfg_builder.cpp#L1212-L1238) for reference.

File: pytorch/tests/ops/comparison/test_eq.py
Comment (Moehre2) [inline pytorch/tests/ops/comparison/test_eq.py:1]: We do not need a special folder for comparisons as we already have one for "pointwise" ops. Rename the test file to `test_comparison.py` and put it into the pointwise test folder.

Also it would be nice if the test is structured similar to the already existing tests.

File: pytorch/tests/ops/comparison/test_eq.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        torch._dynamo.reset()
+        x, y = generate_tensor_pair(shape, dtype)
+        check(EqTensorNet(), x, y, target=target)
+
+    @pytest.mark.parametrize("shape", SHAPES)
+    @pytest.mark.parametrize("dtype", DTYPES)
+    def test_tensor_scalar(
+        self, shape: tuple[int, ...], dtype: torch.dtype, target: str
+    ) -> None:
+        torch._dynamo.reset()
```
Comment (Moehre2) [inline pytorch/tests/ops/comparison/test_eq.py]: Avoid calling `torch._dynamo.reset()` in a test. This is only "allowed" right now for the tests of the models, as they come from a library we do not have control over. Here, we have control over the (simple test) model. So its better to adapt the tests to not require resetting the dynamo cache.

**Review by Moehre2 — CHANGES_REQUESTED** [review]

**Review by Moehre2 — APPROVED** [review]
